apps/compartmentConverter.cpp

```diff
@@ -162,9 +163,22 @@ int main( const int argc, char** argv )
     const brion::CompartmentCounts& counts = in.getCompartmentCounts();
     const brion::GIDSet& gids = in.getGIDs();
 
-    const lunchbox::URI outURI( vm.count( "output" ) == 1 ?
-                                vm[ "output" ].as< std::string >() :
-                                std::string( "out.h5" ));
+    lunchbox::URI outURI( vm.count( "output" ) == 1 ?
+                          vm[ "output" ].as< std::string >() :
+                          std::string( "out.h5" ));
+    if( outURI.getPath().empty( ))
+    {
+        try
+        {
+            outURI.setPath( boost::filesystem::canonical(
+                                inURI.getPath( )).generic_string( ));
+        }
+        catch( const boost::filesystem::filesystem_error& )
+        {
+            // For non-filebased reports, the canonical above will throw.
+            outURI.setPath( inURI.getPath( ));
```

Comment:
If your output is a file, you will naturally give it a path so you won't get here.

If your output is a key-value store, reuse the file-path as the "path" for the KV-store. The KV-store uses the path to uniquefy the keys so that reports don't overwrite each other.

If your input is a file-based report, canonical will make sure that the same file gets written to the same output path. If it's not file-based, the "path" will throw and is then assumed to be "absolute".